[v25.2.x] CORE-8504: Fix KgoVerifierProducer timeout error in CloudRetentionTest

oleiman opened the pull request:
Backport of PR https://github.com/redpanda-data/redpanda/pull/27291 

MANUAL BACKPORT (Conflict in a test that doesn't exist on this branch & some python formatting)

Also pulled in 02a67070881316af77bcfc11068f5603cd1acef0, a minor convenience for in-editor formatting

oleiman committed:
dt/services: Introduce RedpandaServiceABC.wait_until_with_progress_check

Cluster-aware variant of wait_until_with_progress_check that fails out early
if a node dies.

Similar to RedpandaServiceABC.wait_until, checking node liveness each time the
wait condition is evaluated.

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>
(cherry picked from commit d273084aa5a3519b4db5992af58f36874cfdba5b)

Changes:
- Fix formatting

oleiman committed:
dt/kgo: Use progress check in KgoVerifierProducer.wait_for_acks

Optionally via the progress_sec parameter. If not provided falls back to
wait_until.

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>
(cherry picked from commit 253d311f2bd05b351062977de5cec95ac56f62a0)

Conflicts:
- python formatting

oleiman committed:
dt/cloud_retention: Use progress-oriented wait on KgoProducer

We've seen this slow way down in CDT but never stop making progress. Rather
than cranking the timeout even further, enable progress checking when we
wait_for_acks.

Note that we do increase the overall timeout from 10m to 20m, but this is
purposefully over-generous. If we actually stop making progress for some
reason, the underlying waiter should detect that.

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>
(cherry picked from commit 4924b14c497c4863916d783624b0b6881a0d28ba)

Changes:
- Fix formatting

oleiman committed:
dt/util: Simplify wait_until_with_progress_check

Take two zero-param closuers rather than passing the one into the other.

(cherry picked from commit ed57044b54aee20d34689a2d140fead1a7bf1f26)

Conflicts:
- python formatting
- direct_consumer_tests not present on target branch

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--19f90d16-0da4-45ab-a817-43579b5a7240-->
<!--TEST RESULTS-->
<!--2840cb50-426d-436a-8563-7c8c0aafb7de-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/71206">build#71206</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                     |
|:-------------------------|:---------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 2, "compaction_mode": "sliding_window", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/71206#0198cf67-be1d-4def-a6ef-127b1bbb6253 | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72019">build#72019</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                                   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                     |
|:-------------------------|:---------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 2, "compaction_mode": "chunked_sliding_window", "enable_failures": false, "mixed_versions": false, "with_iceberg": false} | integration | https://buildkite.com/redpanda/redpanda/builds/72019#019934db-977e-4b4d-89cb-2ff8a62555b1 | FLAKY         | 19/21    | upstream reliability is '100.0'. current run reliability is '90.47619047619048'. drift is 9.52381 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

oleiman committed:
tools: Break out wrapper script for 'ruff' direct invocation

For usage other than formatting the whole tree. For example formatting a single
file in place.

(cherry picked from commit 02a67070881316af77bcfc11068f5603cd1acef0)

oleiman commented:
/ci-repeat 1
skip-redpanda-build
skip-units

oleiman commented:
/ci-repeat 1
